ocp_resources/datavolume.py

```diff
@@ -216,20 +216,81 @@ def scratch_pvc(self):
             client=self.client,
         )
 
+    def _dv_phase(self):
+        if self.exists:
+            return self.instance.status.phase
+        else:
+            return DV_DOES_NOT_EXISTS
+
     def _check_none_pending_status(self, failure_timeout=120):
         # Avoid waiting for "Succeeded" status if DV's in Pending/None status
         sample = None
         try:
             for sample in TimeoutSampler(
                 wait_timeout=failure_timeout,
                 sleep=10,
-                func=lambda: self.instance.status.phase,
+                func=self._dv_phase,
             ):
                 # If DV status is Pending (or Status is not yet updated) continue to wait, else exit the wait loop
                 if sample in [self.Status.PENDING, None]:
                     continue
                 else:
                     break
         except TimeoutExpiredError:
-            LOGGER.error(f"{self.name} status is {sample}")
+            self.logger.error(f"{self.name} status is {sample}")
             raise
+
+    def is_garbage_collector_enabled_dv(self):
+        if self.exists:
+            dv_annotations = self.instance.metadata.get("annotations")
+            if dv_annotations:
+                return (
+                    dv_annotations.get(f"{Resource.ApiGroup.CDI_KUBEVIRT_IO}/storage.deleteAfterCompletion")
+                    is not False
+                )
+            return True
+        return self.pvc.exists and self.pvc.instance.status.phase == self.pvc.Status.BOUND
+
+    def wait_for_dv_success(self, timeout=600, failure_timeout=120):
+        self.logger.info(f"Wait DV success for {timeout} seconds")
+        self._check_none_pending_status(failure_timeout=failure_timeout)
+        self.pvc.wait_for_status(
+            status=PersistentVolumeClaim.Status.BOUND, timeout=timeout
+        )
+        # if garbage collector enabled, the DV deleted once succeeded
+        if (
+            CDIConfig.is_garbage_collector_enabled_cdi_config()
+            and self.is_garbage_collector_enabled_dv()
+        ):
+            try:
+                for sample in TimeoutSampler(
+                    sleep=1,
+                    wait_timeout=timeout,
+                    func=self._dv_phase,
+                ):
+                    # DV reach to success if the status is succeeded or if the DV does not exists
+                    if sample in [self.Status.SUCCEEDED, DV_DOES_NOT_EXISTS]:
+                        return
+            except TimeoutExpiredError:
+                self.logger.error(f"Status of {self.kind} {self.name} is {sample}")
+                raise
+        else:
+            self.wait_for_status(status=self.Status.SUCCEEDED, timeout=timeout)
+
+    def delete(self, wait=False, timeout=TIMEOUT_4MINUTES, body=None):
+        """
+            Delete DataVolume
```

Comment:
that explain why we need to call different delete() as well

Reply:
ack